src/compiler/checker.ts

```diff
@@ -2989,12 +2983,9 @@ namespace ts {
         }
 
         function typePredicateToString(typePredicate: TypePredicate, enclosingDeclaration?: Declaration, flags?: TypeFormatFlags): string {
-            const writer = getSingleLineStringWriter();

```

Comment:
Can these functions be called multiple times at once? In other words, do they each need to have their own copy of a string writer instead of a singleton?